test/extension.jl

```diff
@@ -36,7 +37,8 @@ end
 
         f_plain = apply_schema(f, sch)
         @test f_plain.rhs.terms[1] isa FunctionTerm
-        @test f_plain == apply_schema(f, sch, Nothing)
+        # this is true but == not defined correctly and apply_schema creates a new instance
+        @test_broken f_plain == apply_schema(f, sch, Nothing)
```

Comment:
What does "this is true" means?

Reply:
they are the same under a reasonable informal definition of "equal" but `==` isn't defined properly (as far as I can remember, it's been some time :joy: )

Reply:
OK. Then maybe this? :-)
```suggestion
        # this works but == is not defined correctly and apply_schema creates a new instance
        @test_broken f_plain == apply_schema(f, sch, Nothing)
```